Added pros and cons of having staking proxy, controller or both (#3935)
w3f/polkadot-wiki: docs/learn/learn-staking-advanced.md

![stash-controller](../assets/stash-controller.png)

Usually, with the stash-controller setup, the stash still needs to sign for actions that are
performed less often, i.e. bonding more funds and changing the controller (see figure above). The
controller is used to sign for those staking actions that are performed more often such as
nominating. Remember, staking on polkadot is not a set-and-forget action, as a nominator you will

> need to monitor the performance of your validators and change them if needed. Also, each time you

Reviewer: ```suggestion
nominating. Remember, staking on polkadot is not a set-and-forget action, as a nominator you will
need to monitor the performance of your validators and make changes if needed. Also, each time you
```

It is not necessary to have a controller if you have a staking proxy: the stash can also be the
controller, and the account security will not be compromised. In this case, the staking proxy will
be used to sign all staking-relate transactions. Note that to change the staking proxy you will need
to sign with the stash.

![stash-stakingProxy](../assets/stash-stakingProxy.png)

This situation is similar to having the controller as a staking proxy. For a practical perspective
we need to use only one account and remember one password to sign for all staking-related
transactions. From a security perspective who controls the staking proxy controls our staking
actions.

It is important to remember that actions that can be performed by the proxy accounts are limited,
and in the case of staking proxy, extrinsic calls to the balances pallet cannot be signed. This
means it is not possible to do balance transfers on the proxied account through a staking proxy.

### Stash is not Controller

If the stash and controller are different accounts, the staking proxy will just be used to bond more
funds and change the controller. Thus the staking proxy will be used to sign for those transactions

> used less often and that are usually signed by the stash.

Reviewer: ```suggestion
that are used less often and that are usually signed by the stash.
```

![stash-controller-stakingProxy](../assets/stash-controller-stakingProxy.png)

From a practical perspective we now have two accounts and we need to remember two passwords. From a
security perspective who wants to fully control our staking actions must have access to two

> accounts.

Reviewer: ```suggestion
From a practical perspective, we now have two accounts and we need to remember two passwords. From a
security perspective, the party who wants to fully control our staking actions must have access to two
accounts.
```

{{ polkadot: DOT is an inflationary token. In fact, there is no maximum number of DOT. Inflation is designed
to be approximately 10% annually, with validator rewards being a function of the amount staked
and the remainder going to the treasury. DOT went through [redenomination](../general/redenomination.md)
in 2020 that saw the DOT token supply increase by 100 times. The current token supply on Polkadot is <RPC network="polkadot" path="query.balances.totalIssuance" defaultValue={12230666300429914781} filter="humanReadable"/> (Over 1.2 Billion DOT). :polkadot }}

{{ kusama: KSM is inflationary; there is no maximum number of KSM. Inflation is designed
to be approximately slightly over 7.5% annually, with validator rewards being a function of the amount staked

> and the remainder going to the treasury. The current token supply on Kusama is
> {{ kusama: <RPC network="kusama" path="query.balances.totalIssuance" defaultValue={12619256191792480093}/> :kusama }}

Reviewer: ```suggestion
and the remainder going to the treasury. The current token supply on Kusama is <RPC network="kusama" path="query.balances.totalIssuance" defaultValue={12619256191792480093}/>
```

This entire paragraph is already wrapped in conditonal rendering so you don't need to wrap the RPC component by itself here as it will actually mess up the formatting.

Before:
![image](https://user-images.githubusercontent.com/13341935/196609550-eea87d5e-f6b1-4910-896f-1a61bc210c50.png)

After:
![image](https://user-images.githubusercontent.com/13341935/196609651-5f7c45af-0cda-4eaf-a112-ab7b9d1d61ec.png)

:::info The ideal staking rate varies based on the number of parachains

The current staking rate on Polkadot still assumes the absence of parachains, with the suggested
ideal staking rate of 75%. You can track the progress on the issue to adjust it
[here](https://github.com/paritytech/polkadot/pull/5872). This has already been adjusted on Kusama.

> At the time of updating this section, the ideal staking rate on Kusama was 50.5% when there were 49

Reviewer: Maybe use a date here instead for clarity.

Author: Now that I think about it, probably better to remove this dependence on the date of writing the section. I rewrote  it in more general terms.
